torch/_dynamo/trace_rules.py

```diff
@@ -35,40 +40,2732 @@
+    "math.acos": TorchInGraphFunctionVariable,
+    "math.acosh": TorchInGraphFunctionVariable,
+    "math.asin": TorchInGraphFunctionVariable,
+    "math.asinh": TorchInGraphFunctionVariable,
+    "math.atan": TorchInGraphFunctionVariable,
+    "math.atan2": TorchInGraphFunctionVariable,
+    "math.atanh": TorchInGraphFunctionVariable,
+    "math.ceil": TorchInGraphFunctionVariable,
+    "math.copysign": TorchInGraphFunctionVariable,
+    "math.cos": TorchInGraphFunctionVariable,
+    "math.cosh": TorchInGraphFunctionVariable,
+    "math.degrees": TorchInGraphFunctionVariable,
+    "math.dist": TorchInGraphFunctionVariable,
+    "math.erf": TorchInGraphFunctionVariable,
+    "math.erfc": TorchInGraphFunctionVariable,
+    "math.exp": TorchInGraphFunctionVariable,
+    "math.expm1": TorchInGraphFunctionVariable,
+    "math.fabs": TorchInGraphFunctionVariable,
+    "math.factorial": TorchInGraphFunctionVariable,
+    "math.floor": TorchInGraphFunctionVariable,
+    "math.fmod": TorchInGraphFunctionVariable,
+    "math.frexp": TorchInGraphFunctionVariable,
+    "math.fsum": TorchInGraphFunctionVariable,
+    "math.gamma": TorchInGraphFunctionVariable,
+    "math.gcd": TorchInGraphFunctionVariable,
+    "math.hypot": TorchInGraphFunctionVariable,
+    "math.isclose": TorchInGraphFunctionVariable,
+    "math.isfinite": TorchInGraphFunctionVariable,
+    "math.isinf": TorchInGraphFunctionVariable,
+    "math.isnan": TorchInGraphFunctionVariable,
+    "math.isqrt": TorchInGraphFunctionVariable,
+    "math.lcm": TorchInGraphFunctionVariable,
+    "math.ldexp": TorchInGraphFunctionVariable,
+    "math.lgamma": TorchInGraphFunctionVariable,
+    "math.log": TorchInGraphFunctionVariable,
+    "math.log1p": TorchInGraphFunctionVariable,
+    "math.log10": TorchInGraphFunctionVariable,
+    "math.log2": TorchInGraphFunctionVariable,
+    "math.modf": TorchInGraphFunctionVariable,
+    "math.pow": TorchInGraphFunctionVariable,
+    "math.radians": TorchInGraphFunctionVariable,
+    "math.remainder": TorchInGraphFunctionVariable,
+    "math.sin": TorchInGraphFunctionVariable,
+    "math.sinh": TorchInGraphFunctionVariable,
+    "math.tan": TorchInGraphFunctionVariable,
+    "math.tanh": TorchInGraphFunctionVariable,
+    "math.trunc": TorchInGraphFunctionVariable,
+    "math.prod": TorchInGraphFunctionVariable,
+    "math.perm": TorchInGraphFunctionVariable,
+    "math.comb": TorchInGraphFunctionVariable,
+    "math.nextafter": TorchInGraphFunctionVariable,
+    "math.ulp": TorchInGraphFunctionVariable,
 }
 
 
+torch_name_rule_map = {**manual_torch_name_rule_map, **auto_torch_name_rule_map}
+
+
 @functools.lru_cache(None)
 def get_torch_obj_rule_map():
     d = dict()
     for k, v in torch_name_rule_map.items():
-        obj = load_object(k)
-        assert obj not in d
-        d[obj] = v
+        try:
+            obj = load_object(k)
+            d[obj] = v
+        except Exception:
+            pass
```

Comment:
The try ... except is to handle no CUDA functions in CPU only env, probably I should have a better way. But let me check if the idea is in right direction in case wasting time on trivial things.

Reply:
Let's make the `except` case more narrow here.

Reply:
Yes